Sources/StreamChatUI/Common Views/BaseViews.swift

```diff
@@ -274,46 +195,69 @@ open class ViewController: UIViewController, AppearanceSetting, Customizable {
     }
     
     public func defaultAppearance() { /* default empty implementation */ }
-    open func setUp() { /* default empty implementation */ }
+    open func setUp() { TraitCollectionWatcher.activate() }
     open func setUpAppearance() { /* default empty implementation */ }
     open func setUpLayout() { /* default empty implementation */ }
     open func updateContent() { /* default empty implementation */ }
+    open func resetAppearance() {
+        (self as! Self).applyDefaultAppearance()
+        setUpAppearance()
+        updateContent()
+    }
+}
 
-    override open func traitCollectionDidChange(_ previousTraitCollection: UITraitCollection?) {
-        super.traitCollectionDidChange(previousTraitCollection)
+/// Observes trait collection changes and calls `Customizable.resetAppearance` in child-first DFS way
+/// on view & view controller hierarchies.
+private enum TraitCollectionWatcher {
+    private class Wiretap: View {
+        var onTraitChange: () -> Void = {}
+        override func setUp() { /* do not call super to prevent loop */ } // swiftlint:disable:this overridden_super_call
+        override open func traitCollectionDidChange(_ previousTraitCollection: UITraitCollection?) {
+            super.traitCollectionDidChange(previousTraitCollection)
+
+            guard #available(iOS 12, *) else { return }
+            guard previousTraitCollection?.userInterfaceStyle != traitCollection.userInterfaceStyle else { return }
+            onTraitChange()
+        }
+    }
 
-        guard #available(iOS 12, *) else { return }
-        guard previousTraitCollection?.userInterfaceStyle != traitCollection.userInterfaceStyle else { return }
+    private static var activeTap: Wiretap!
 
-        // Execute previous pending updates
-        pendingUpdates?()
+    static func activate() {
+        guard activeTap == nil else { return }
 
-        // Add this view's updates to pending updates
-        pendingUpdates = {
-            (self as! Self).applyDefaultAppearance()
-            self.setUpAppearance()
-            self.updateContent()
+        activeTap = Wiretap(frame: CGRect(x: -10, y: -10, width: 1, height: 1))
+        activeTap.backgroundColor = .clear
+        activeTap.onTraitChange = {
+            DispatchQueue.main.async {
+                TraitCollectionWatcher.dfsAppearanceReset()
+            }
         }
 
-        // Make sure all pending updates are executed eventually
-        DispatchQueue.main.async {
-            pendingUpdates?()
-            pendingUpdates = nil
+        UIApplication.shared.keyWindow?.addSubview(activeTap)
```

Comment:
So... this view will last forever until application is shut down. I have few questions :D 

1. What are the drawbacks of this? 
2. Doesn't it break something?

Reply:
Unless someone heavily rely on special subviews order in active window (why would you?), it shouldn't matter.